doc/source/guide/derivatives_guide.rst

```diff
@@ -0,0 +1,151 @@
+.. _derivatives_in_depth:
+
+##############################################
+A guide to the different notions of derivative
+##############################################
+
+The concept of a derivative is one of the core concepts of mathematical analysis and is used where we want a linear approximation of an arbitrary function. Since the derivative is used in different contexts to mean slightly different things, this guide has been written to clarify and introduce what the different concepts mean.
+
+In short, different notions of derivatives that will be discussed here are:
+
+* **Derivative**. When we write derivative in ODL code and documentation, we intend the derivative of a `Operator` :math:`A : \mathcal{X} \rightarrow \mathcal{Y}` w.r.t to a disturbance in :math:`x`, i.e a linear approximation of :math:`A(x + h)` for small :math:`h`. The derivative in a point :math:`x` is an operator :math:`A'(x) : \mathcal{X} \rightarrow \mathcal{Y}`.
+
+* **Gradient**. If the operator :math:`A` is a `functional`, i.e. :math:`A : \mathcal{X} \rightarrow \mathbb{R}`, then the gradient is a measure of how much the output would change w.r.t. a small change in the input. The gradient in a point :math:`x` is a vector :math:`[\nabla A](x)` in :math:`\mathcal{X}` such that :math:`A'(x)(y) = \langle [\nabla A](x), y \rangle`. The gradient operator is the operator :math:`x \rightarrow [\nabla A](x)`.
+
+* **Hessian**. The hessian in a point :math:`x` is the derivative operator of the gradient operator, i.e. :math:`H(x) = [\nabla A]'(x)`.
+
+* **Spatial Gradient**. The spatial gradient is only defined for spaces :math:`\mathcal{F}(\Omega, \mathbb{F})` whose elements are functions over some domain :math:`\Omega \subset \mathbb{R}^n` with range in :math:`\mathbb{R}` or :math:`\mathbb{C}` . It can be seen as a vectorized version of the usual gradient, taken in each point in :math:`\Omega`.
+
+Derivative
+##########
+
+The derivative is usually introduced for functions :math:`f: \mathbb{R} \rightarrow \mathbb{R}` via the limit
+
+.. math::
+    f'(x) = \lim_{h \rightarrow 0} \frac{f(x + h) - f(x)}{h}
+
+Here we say that the derivative of :math:`f` in :math:`x` is :math:`f'(x)`.
+
+This limit makes sense in one dimension, but once we start considering functions in higher dimension we get into trouble.
+Consider :math:`f: \mathbb{R}^n \rightarrow \mathbb{R}^m`, what would :math:`h` mean in this case?
+A extension is the concept of a directional derivative. The derivative of :math:`f` in :math:`x` in *direction* :math:`d` is :math:`f'(x)(d)`.
+
+.. math::
+    f'(x)(d) = \lim_{h \rightarrow 0} \frac{f(x + dh) - f(x)}{h}
+
+Here we see (as implied by the notation) that :math:`f'(x)` is actually an operator
+
+.. math::
+    f'(x) : \mathbb{R}^n \rightarrow \mathbb{R}^m
```

Comment:
Add either a dot at the end, or an extra half sentence like (I'm making up notation now) "that takes a vector `d \in R^n` and maps it to the vector `f'(x)(d) \in R^m`".